packages/qwik/src/components/Form.tsx

```diff
@@ -9,16 +9,24 @@ import type {
   FieldArrayPath,
 } from '../types';
 
+export type FormSubmitHandler<TFieldValues extends FieldValues> = PropFunction<
+  (values: TFieldValues, event: QwikSubmitEvent<HTMLFormElement>) => unknown
+>;
+
+export type FormAction<TFieldValues extends FieldValues> = ActionStore<
+  any,
+  TFieldValues,
+  true
+>; // TODO: Improve generics
+
 export type FormProps<
   TFieldValues extends FieldValues,
   TFieldName extends FieldPath<TFieldValues>,
   TFieldArrayName extends FieldArrayPath<TFieldValues>
 > = Omit<QwikJSX.IntrinsicElements['form'], 'action' | 'method'> & {
   of: FormStore<TFieldValues, TFieldName, TFieldArrayName>;
-  action?: ActionStore<any, TFieldValues, true>; // TODO: Improve generics
-  onSubmit$?: PropFunction<
-    (values: TFieldValues, event: QwikSubmitEvent<HTMLFormElement>) => unknown
-  >;
+  action?: FormAction<TFieldValues>;
```

Comment:
Can you undo this line and remove the `FormAction` type above? I am currently in the middle of integrating the actions and this type has changed in the meantime.

Reply:
I removed the FormAction type and changed it back to how it was.